--- ./AlgorithmArray_orig.def.h	2022-06-16 14:42:37.978886843 -0700
+++ src/Parallel/Algorithms/AlgorithmArray.def.h	2022-06-16 14:43:04.995671535 -0700
@@ -550,2 +550,2 @@
-  _TRACE_CREATION_DETAILED(&env, CkIndex_AlgorithmArray <ParallelComponent, SpectreArrayIndex> ::template idx_invoke_iterable_action_void<ThisAction, PhaseIndex, DataBoxIndex>());
+  _TRACE_CREATION_DETAILED(&env, (CkIndex_AlgorithmArray <ParallelComponent, SpectreArrayIndex> ::template idx_invoke_iterable_action_void<ThisAction, PhaseIndex, DataBoxIndex>()));
   _TRACE_CREATION_DONE(1);
