diff --git a/cmake/detect-features-cxx.cmake b/cmake/detect-features-cxx.cmake
index d3aa6ab94..830da9cf1 100644
--- a/cmake/detect-features-cxx.cmake
+++ b/cmake/detect-features-cxx.cmake
@@ -38,12 +38,12 @@ endif()
 
 # Needed so that tlsglobals works correctly with --build-shared
 # See https://github.com/UIUC-PPL/charm/issues/3168 for details.
-check_cxx_compiler_flag("-ftls-model=initial-exec" CMK_COMPILER_KNOWS_FTLS_MODEL)
-if(CMK_COMPILER_KNOWS_FTLS_MODEL)
-  set(OPTS_CC "${OPTS_CC} -ftls-model=initial-exec")
-  set(OPTS_CXX "${OPTS_CXX} -ftls-model=initial-exec")
-  set(OPTS_LD "${OPTS_LD} -ftls-model=initial-exec")
-endif()
+# check_cxx_compiler_flag("-ftls-model=initial-exec" CMK_COMPILER_KNOWS_FTLS_MODEL)
+# if(CMK_COMPILER_KNOWS_FTLS_MODEL)
+#   set(OPTS_CC "${OPTS_CC} -ftls-model=initial-exec")
+#   set(OPTS_CXX "${OPTS_CXX} -ftls-model=initial-exec")
+#   set(OPTS_LD "${OPTS_LD} -ftls-model=initial-exec")
+# endif()
 
 # Allow seeing own symbols dynamically, needed for programmatic backtraces
 check_cxx_compiler_flag("-rdynamic" CMK_COMPILER_KNOWS_RDYNAMIC)
diff --git a/src/scripts/configure.ac b/src/scripts/configure.ac
index f6c0f311b..41a7c9f46 100644
--- a/src/scripts/configure.ac
+++ b/src/scripts/configure.ac
@@ -820,15 +820,15 @@ then
 fi
 
 # Needed so that tlsglobals works correctly with --build-shared
-# See https://github.com/UIUC-PPL/charm/issues/3168 for details.
-test_cxx "whether C++ compiler accepts -ftls-model=initial-exec" "yes" "no" "-ftls-model=initial-exec"
-if test $strictpass -eq 1
-then
-    add_flag 'CMK_COMPILER_KNOWS_FTLS_MODEL="1"' "tlsglobals"
-    OPTS_CC="$OPTS_CC -ftls-model=initial-exec"
-    OPTS_CXX="$OPTS_CXX -ftls-model=initial-exec"
-    OPTS_LD="$OPTS_LD -ftls-model=initial-exec"
-fi
+# # See https://github.com/UIUC-PPL/charm/issues/3168 for details.
+# test_cxx "whether C++ compiler accepts -ftls-model=initial-exec" "yes" "no" "-ftls-model=initial-exec"
+# if test $strictpass -eq 1
+# then
+#     add_flag 'CMK_COMPILER_KNOWS_FTLS_MODEL="1"' "tlsglobals"
+#     OPTS_CC="$OPTS_CC -ftls-model=initial-exec"
+#     OPTS_CXX="$OPTS_CXX -ftls-model=initial-exec"
+#     OPTS_LD="$OPTS_LD -ftls-model=initial-exec"
+# fi
 
 # Test for a flag important for shared linking
 test_cxx "whether C++ compiler accepts -fvisibility=hidden" "yes" "no" "-fvisibility=hidden"
